deployments/datahub/images/default/environment.yml

```diff
@@ -29,4 +26,248 @@ dependencies:
+- PyPDF2==1.26.0
+
+# data100 scientific packages
+- xlrd==2.0.1
+
+# data100 visualization
+- colorlover==0.3.0
+- cufflinks-py==0.17.3
+- altair==4.1.0
+
+# data100 tools to access things
+- tweepy==3.10.0
+- pytz==2021.1
+- psycopg2==2.9.1
+
+# data102
+- distributed==2021.8.0
+- keras-applications==1.0.8
+- keras-preprocessing==1.1.2
+- keras==2.6.0
+- cytoolz==0.11.0
+- pyro-ppl==1.7.0
+- lime==0.2.0.1
+- shap==0.39.0
+
+# prob140 2021 Spring
+- sympy==1.8
+
+# ESPM 167 - https://github.com/berkeley-dsep-infra/datahub/issues/2278
+- contextily==1.1.0
+
+# JupyterLab pypi extensions
+- jupyterlab-geojson==3.1.2
+
+# data100, fall 2021 - https://github.com/berkeley-dsep-infra/datahub/issues/2875
+- metpy==1.1.0
+- pooch==1.5.2
+
+# PS88 https://github.com/berkeley-dsep-infra/datahub/issues/2925
+- linearmodels==4.24
+
+# https://github.com/berkeley-dsep-infra/datahub/issues/2950
+# Needed to work with a new enough version of httplib2
+- httplib2>=0.20.2
+
+- pip:
+# data8; foundation
+  - datascience==0.17.0
+  - okpy==1.18.1
+  - folium==0.12.1
+  
+# r
+  - jupyter-shiny-proxy==1.1
+  
+# modules
+  - nb2pdf==0.6.2
+  
+# phys 151
+  - daft==0.1.2
+  
+# ce88
+  - pymdptoolbox==4.0-b3
+
+# data-x; DL
+  - tensorflow==2.6.*
+  - opencv-python==4.5.3.56
+  - scikit-image==0.18.*
+  
+# astr 128/256; spring 2021
+  - exoplanet==0.5.1
+  - torch==1.9.0
+  - dustmaps==1.0.9
+
+# Google spreadsheets, Eric Van Dusen / Keeley Takimoto / Modules
+  - gspread-pandas==2.3.0
+  
+# data 88e; spring 2021
+  - csaps==1.0.4
+  - nbforms==0.5.1
+  
+# espm 125/bio 105; fall 2019
+# see https://github.com/berkeley-dsep-infra/datahub/issues/1796
+  - nlmpy==1.0.1
+  
+# physics 188/288 fall, 2019
+  - getdist==1.3.1
+  - tensorflow-hub==0.12.0
+  - tensorflow-probability==0.13.0
+
+# mcb 163l, fall 2019
+  - allensdk==2.12.2
+  
+# cs16A/B, spring 2020
+  - lcapy==0.96
+  
+# ds198 mch infodemiology, fall 2020/spring 2021
+# google apis
+  - google==3.0.0
+  
+# data100 scientific packages
+  - ray==1.5.2
+  
+# data100 teaching
+  - jassign==0.0.7
+  - dsassign==0.0.8
+  
+# data102
+  - keras-vis==0.4.1
+  - plotly-express==0.4.1
+  - dask==2021.8.0
+  
+# prob140 2021 Spring
+  - prob140==0.4.1.5
+  
+# INDENG 142 Spring 2021 (future semesters as well) - https://github.com/berkeley-dsep-infra/datahub/issues/2314
+  - fancyimpute==0.6.0
+  
+# IB 105 / ESPM 125, Fall 2021 - https://github.com/berkeley-dsep-infra/datahub/issues/2696
+  - geonomics==1.3.0
+  
+# data100, fall 2021 - https://github.com/berkeley-dsep-infra/datahub/issues/2767
+  - ipython-sql==0.4.0
+  
+# LBL summer research internship in materials informatics. Jupyter Book curriculum at https://enze-chen.github.io/mi-book/
+  - pymatgen==2022.0.11
+  - matminer==0.7.4
+  
+# ls 88-3; neuro
+  - mne==0.23.0
+
+# data100, geog88
+# https://github.com/berkeley-dsep-infra/datahub/issues/2838
+  - pysal==2.5.*
+  
+# cogsci88;
+  - ggplot==0.11.5
+
+# pymc3 needs this
+  - mkl-service==2.4.0
```

Comment:
https://github.com/berkeley-dsep-infra/datahub/pull/2544 explicitly installed this from conda-forge, and I think we should keep it there. mkl (https://en.wikipedia.org/wiki/Math_Kernel_Library) is a non-python math library, and the description on https://pypi.org/project/mkl-service/ says to install from conda. Can you try importing the package from the built image? If that works maybe we're clear...

Reply:
@yuvipanda sadly it does not work.

So I'm moving mkl-service back to conda and doing more trial and error and dependency juggling.